website/docs/r/upgrade_run.html.markdown

```diff
@@ -0,0 +1,105 @@
+---
+subcategory: "Beta"
+layout: "nsxt"
+page_title: "NSXT: nsxt_upgrade_run"
+description: A resource to configure and execute upgrade of NSXT cluster.
+---
+
+# nsxt_upgrade_run
+
+This resource provides a method to configure and execute upgrade of NSXT cluster.
+
+It will first configure upgrade unit groups and upgrade plan settings for `EDGE`
+and `HOST` components. Then it will execute the upgrade for each component. If 
+there are either `EDGE` or `HOST` groups disabled from the upgrade, the corresponding 
+component will be paused after enabled groups get upgraded. At the same time, `MP` upgrade
+won't be processed, because that requires `EDGE` and `HOST` to be fully upgraded. Refer to
+the exposed attribute for current upgrade state details.
+
+If upgrade post-checks are configured to be run, it will trigger the upgrade post-check.
+Please use data source `nsxt_upgrade_postcheck` to retrieve results of upgrade post-checks.
+
+## Example Usage
+
+```hcl
+resource "nsxt_upgrade_run" "run1" {
+  upgrade_prepare_id = nsxt_upgrade_prepare.test.id
+
+  edge_group {
+    id      = data.nsxt_edge_upgrade_group.eg1.id
+    enabled = false
+  }
+
+  edge_group {
+    id      = data.nsxt_edge_upgrade_group.eg2.id
+    enabled = true
+  }
+
+  host_group {
+    id       = data.nsxt_edge_upgrade_group.eg2.id
+    parallel = true
+  }
+
+  edge_upgrade_setting {
+    parallel           = true
+    post_upgrade_check = true
+  }
+
+  host_upgrade_setting {
+    parallel           = true
+    post_upgrade_check = true
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `upgrade_prepare_ready_id` - (Required) ID of corresponding `nsxt_upgrade_prepare_ready` resource. Update this field will trigger replacement (destroy and create) of this resource.
+* `edge_group` - (Optional) EDGE component upgrade unit group configurations. Groups will be reordered following the order they present in this field.
+    * `id` - (Required) ID of the upgrade unit group.
+    * `enabled` - (Optional) Flag to indicate whether upgrade of this group is enabled or not. Default: True.
+    * `parallel` - (Optional) Upgrade method to specify whether the upgrade is to be performed in parallel or serially. Default: True.
+    * `pause_after_each_upgrade_unit` - (Optional) Flag to indicate whether upgrade should be paused after upgrade of each upgrade-unit. Default: False.
```

Comment:
For that component, it will be paused. The MP upgrade won't be processed, which requires EDGE and HOST upgrade to be successful. If a user wants to resume the upgrade he could re-apply this resource to trigger the upgrade.